feat: add C `ndarray` API and refactor for `stats/base/sdsnanmeanors`

DhruvArvindSingh opened the pull request:
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds C `ndarray` implementation for `stats/base/sdsnanmeanors`.
-   Refactors `stats/base/sdsnanmeanors`.
-   Adds C implementation section in README.md.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

DhruvArvindSingh committed:
feat: add C ndarray and refactor for stats/base/sdsnanmeanors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

DhruvArvindSingh commented:
/stdlib update-copyright-years

DhruvArvindSingh committed:
Merge branch 'stdlib-js:develop' into feat/ndarray_sdsnanmeanors

DhruvArvindSingh committed:
refactor: refactored benchmarks, tests and docs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/ndarray_sdsnanmeanors

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/ndarray_sdsnanmeanors

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -51,7 +51,7 @@ The [arithmetic mean][arithmetic-mean] is defined as
-#### sdsnanmeanors( N, x, stride )
+#### sdsnanmeanors( N, x, strideX )
 
 Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using ordinary recursive summation with extended accumulation.
```
```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using ordinary recursive summation with extended accumulation.
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -69,18 +69,16 @@ The function has the following parameters:
 -   **N**: number of indexed elements.
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
--   **stride**: index increment for `x`.
+-   **strideX**: index increment for `x`.
```
```suggestion
-   **strideX**: stride length for `x`.
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using ordinary recursive summation with extended accumulation.
+
+```c
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
````
```suggestion
const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +163,109 @@ console.log( v );
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_sdsnanmeanors( x.length, x, 1 );
+// returns ~0.3333
```
```suggestion
// returns ~0.3333f
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
+
+```c
+double stdlib_strided_sdsnanmeanors( const CBLAS_INT N, const double *X, const CBLAS_INT strideX );
````
```suggestion
float stdlib_strided_sdsnanmeanors( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using ordinary recursive summation with extended accumulation and alternative indexing semantics.
+
+```c
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
````
```suggestion
const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +163,109 @@ console.log( v );
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_sdsnanmeanors_ndarray( x.length, x, 1, 0 );
+// returns ~0.3333
```
```suggestion
// returns ~0.3333f
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+```c
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_sdsnanmeanors( x.length, x, 1 );
````
```suggestion
float v = stdlib_strided_sdsnanmeanors( x.length, x, 1 );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+```c
+const float x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_sdsnanmeanors_ndarray( x.length, x, 1, 0 );
````
```suggestion
float v = stdlib_strided_sdsnanmeanors_ndarray( x.length, x, 1, 0 );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
````diff
@@ -181,6 +163,109 @@ console.log( v );
+-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.
+
+```c
+double stdlib_strided_sdsnanmeanors_ndarray( const CBLAS_INT N, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
````
```suggestion
float stdlib_strided_sdsnanmeanors_ndarray( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +163,109 @@ console.log( v );
+
+int main( void ) {
+    // Create a strided array:
+    const float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
```suggestion
    const float x[] = { 1.0f, 2.0f, 0.0f/0.0f, 3.0f, 0.0f/0.0f, 4.0f, 5.0f, 6.0f, 0.0f/0.0f, 7.0f, 8.0f, 0.0f/0.0f };
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +163,109 @@ console.log( v );
+    const int strideX = 2;
+
+    // Compute the arithmetic mean:
+    double v = stdlib_strided_sdsnanmeanors( N, x, strideX );
```
```suggestion
    float v = stdlib_strided_sdsnanmeanors( N, x, strideX );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +163,109 @@ console.log( v );
+    double v = stdlib_strided_sdsnanmeanors( N, x, strideX );
+
+    // Print the result:
+    printf( "mean: %lf\n", v );
```
```suggestion
    printf( "mean: %f\n", v );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/examples/c/example.c:
```diff
@@ -17,21 +17,20 @@
 int main( void ) {
 	// Create a strided array:
-	float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
+	const float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
```suggestion
	const float x[] = { 1.0f, 2.0f, 0.0f/0.0f, 3.0f, 0.0f/0.0f, 4.0f, 5.0f, 6.0f, 0.0f/0.0f, 7.0f, 8.0f, 0.0f/0.0f };
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/include/stdlib/stats/base/sdsnanmeanors.h:
```diff
@@ -31,7 +31,12 @@ extern "C" {
 * Computes the arithmetic mean of a single-precision floating-point strided array, ignoring `NaN` values and using ordinary recursive summation with extended accumulation.
 */
-float stdlib_strided_sdsnanmeanors( const int64_t N, const float *X, const int64_t stride );
+double API_SUFFIX( stdlib_strided_sdsnanmeanors )( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
```
```suggestion
float API_SUFFIX(stdlib_strided_sdsnanmeanors)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/include/stdlib/stats/base/sdsnanmeanors.h:
```diff
@@ -31,7 +31,12 @@ extern "C" {
+/**
+* Computes the arithmetic mean of a single-precision floating-point strided array, ignoring `NaN` values and using ordinary recursive summation with extended accumulation and alternative indexing semantics.
+*/
+double API_SUFFIX( stdlib_strided_sdsnanmeanors_ndarray )( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```
```suggestion
float API_SUFFIX(stdlib_strided_sdsnanmeanors_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/ndarray.native.js:
```diff
@@ -20,38 +20,30 @@
-* var v = sdsnanmeanors( N, x, 2, 1 );
-* // returns 1.25
+* var v = sdsnanmeanors( 4, x, 2, 1 );
+*  //returns 1.25
```
```suggestion
* //returns 1.25
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/sdsnanmeanors.js:
```diff
@@ -30,50 +31,20 @@ var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
-	}
-	return float64ToFloat32( sum / n );
+function sdsnanmeanors( N, x, strideX ) {
+	var ox = stride2offset( N, strideX );
```
```suggestion
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/sdsnanmeanors.js:
```diff
@@ -30,50 +31,20 @@ var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
-	return float64ToFloat32( sum / n );
+function sdsnanmeanors( N, x, strideX ) {
+	var ox = stride2offset( N, strideX );
+	return ndarray( N, x, strideX, ox );
```
```suggestion
	return ndarray( N, x, strideX, stride2offset( N, strideX ) );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/sdsnanmeanors.native.js:
```diff
@@ -30,20 +30,19 @@ var addon = require( './../src/addon.node' );
-* var v = sdsnanmeanors( N, x, 1 );
-* // returns ~0.3333
+* var v = sdsnanmeanors( x.length, x, 1 );
+*  //returns ~0.3333
```
```suggestion
* //returns ~0.3333
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/addon.c:
```diff
@@ -34,10 +35,27 @@
-	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_sdsnanmeanors( N, X, stride ), v );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX( stdlib_strided_sdsnanmeanors )( N, X, strideX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_sdsnanmeanors)( N, X, strideX ), v );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/addon.c:
```diff
@@ -34,10 +35,27 @@
+	STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX( stdlib_strided_sdsnanmeanors_ndarray )( N, X, strideX, offsetX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_sdsnanmeanors_ndarray)( N, X, strideX, offsetX ), v );
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -17,43 +17,54 @@
+* @return         output value
 */
-float stdlib_strided_sdsnanmeanors( const int64_t N, const float *X, const int64_t stride ) {
+double API_SUFFIX( stdlib_strided_sdsnanmeanors )( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
```
```suggestion
float API_SUFFIX(stdlib_strided_sdsnanmeanors)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -17,43 +17,54 @@
+* @param offsetX  starting index for X
+* @return         output value
+*/
+double API_SUFFIX( stdlib_strided_sdsnanmeanors_ndarray )( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
```
```suggestion
float API_SUFFIX(stdlib_strided_sdsnanmeanors_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/test/test.sdsnanmeanors.js:
```diff
@@ -62,7 +61,7 @@ tape( 'the function calculates the arithmetic mean of a strided array, ignoring
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN`', function test( t ) {
+tape( 'if provided an `4` parameter less than or equal to `0`, the function returns `NaN`', function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN`', function test( t ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/test/test.sdsnanmeanors.js:
```diff
@@ -77,7 +76,7 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
 });
 
-tape( 'if provided an `N` parameter equal to `1`, the function returns the first element', function test( t ) {
+tape( 'if provided an `4` parameter equal to `1`, the function returns the first element', function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter equal to `1`, the function returns the first element', function test( t ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/test/test.sdsnanmeanors.native.js:
```diff
@@ -153,7 +152,7 @@ tape( 'the function calculates the arithmetic mean of a strided array, ignoring
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN`', opts, function test( t ) {
+tape( 'if provided an `4` parameter less than or equal to `0`, the function returns `NaN`', opts, function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN`', opts, function test( t ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/test/test.sdsnanmeanors.native.js:
```diff
@@ -168,7 +167,7 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
 });
 
-tape( 'if provided an `N` parameter equal to `1`, the function returns the first element', opts, function test( t ) {
+tape( 'if provided an `4` parameter equal to `1`, the function returns the first element', opts, function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter equal to `1`, the function returns the first element', opts, function test( t ) {
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/manifest.json:
```diff
@@ -28,7 +28,7 @@
     {
```
needs to be updated

gururaj1512 committed:
chore: apply code suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
style: fix indentations

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

gururaj1512 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feat/ndarray_sdsnanmeanors

gururaj1512 committed:
fix: examples and benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: minor cleanup

gururaj1512 committed:
fix: simplify example usage in README

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -51,36 +51,33 @@ The [arithmetic mean][arithmetic-mean] is defined as
-var N = floor( x.length / 2 );
 
-var v = sdsnanmeanors( N, x, 2 );
+var v = sdsnanmeanors( 4, x, 2 );
```
```suggestion
var v = sdsnanmeanors( 5, x, 2 );
```

kgryte committed:
docs: update example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +169,109 @@ console.log( v );
+<!-- /.c -->
+
+* * *
+
```
```suggestion
```

kgryte committed:
docs: remove horizontal rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +169,107 @@ console.log( v );
 
 <!-- /.examples -->
 
+<!-- C usage documentation. -->
```
```suggestion
* * *

<!-- C usage documentation. -->
```

kgryte committed:
docs: add horizontal rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/README.md:
```diff
@@ -181,6 +169,109 @@ console.log( v );
 
 <!-- /.examples -->
 
+* * *
+
+<!-- C usage documentation. -->
```
```suggestion
<!-- C usage documentation. -->

* * *
```

kgryte committed:
style: move rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/docs/repl.txt:
```diff
@@ -37,22 +37,19 @@
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
-    > var stride = 2;
-    > {{alias}}( N, x, stride )
+    > {{alias}}( 3, x, 2 )
```
```suggestion
    > {{alias}}( 4, x, 2 )
```

kgryte committed:
docs: update example

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/docs/repl.txt:
```diff
@@ -69,10 +66,10 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/docs/repl.txt:
```diff
@@ -22,8 +22,8 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/ndarray.native.js:
```diff
@@ -20,38 +20,30 @@
-* @param {integer} stride - stride length
-* @param {NonNegativeInteger} offset - starting index
+* @param {integer} strideX - stride length
+* @param {NonNegativeInteger} offsetX - starting index for x
```
```suggestion
* @param {NonNegativeInteger} offsetX - starting index
```

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/ndarray.native.js:
```diff
@@ -20,38 +20,30 @@
-* var v = sdsnanmeanors( N, x, 2, 1 );
-* // returns 1.25
+* var v = sdsnanmeanors( 4, x, 2, 1 );
+* //returns 1.25
```
```suggestion
* // returns 1.25
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/lib/sdsnanmeanors.native.js:
```diff
@@ -30,20 +30,19 @@ var addon = require( './../src/addon.node' );
-* var v = sdsnanmeanors( N, x, 1 );
-* // returns ~0.3333
+* var v = sdsnanmeanors( x.length, x, 1 );
+* //returns ~0.3333
```
```suggestion
* // returns ~0.3333
```

kgryte committed:
style: fix missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -0,0 +1,73 @@
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  stride length
+* @param offsetX  starting index for X
```
```suggestion
* @param offsetX  starting index
```

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -0,0 +1,73 @@
+	const CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX( stdlib_strided_sdsnanmeanors_ndarray )( N, X, strideX, ox );
+}
+/**
```
```suggestion

/**
```

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -0,0 +1,73 @@
+*/
+float API_SUFFIX(stdlib_strided_sdsnanmeanors)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
+	const CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX( stdlib_strided_sdsnanmeanors_ndarray )( N, X, strideX, ox );
```
```suggestion
	return API_SUFFIX(stdlib_strided_sdsnanmeanors_ndarray)( N, X, strideX, ox );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/sdsnanmeanors/src/main.c:
```diff
@@ -0,0 +1,74 @@
+* @return         output value
+*/
+float API_SUFFIX(stdlib_strided_sdsnanmeanors_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
+	double sum;
+	CBLAS_INT ix;
+	CBLAS_INT i;
+	CBLAS_INT n;
```
```suggestion
	CBLAS_INT ix;
	CBLAS_INT i;
	CBLAS_INT n;
	double sum;
```

kgryte committed:
style: reorder declarations

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after clean-up.
